posts/atomic-design-in-practice.md

```diff
@@ -0,0 +1,233 @@
+
+In the example, we have:
+
+* Search form
+* Main navigation
+
+![Molecules: search form, main navigation](https://raw.githubusercontent.com/Gnuk/blog-usa/atomic-design-in-practice/images/2019/10/atomic-molecules.png)
+
+## Form by example
+
+The **search form** molecule positions a **label** atom above a **field** atom followed by a **search button** atom.
+
+### HTML (Pug)
+
+```pug
+.molecule-form-search
+    .molecule-form-search-label
+        +atom-label
+    .molecule-form-search-fields
+        .molecule-form-search-input
+            +atom-input
+        .molecule-form-search-search
+            +atom-search
+```
+
+### CSS (SCSS)
+
+```scss
+.molecule-form-search {
+  .molecule-form-search-fields {
+    display: flex;
+    flex-wrap: nowrap;
+    .molecule-form-search-input {
+      margin-right: 8px;
+      flex-grow: 1;
+      input.atom-input {
+        width: 100%;
+      }
+    }
+  }
+}
+```
+
+# Organisms
+
+Organisms arrange molecules and atils together, in our case, it's the **header**.
+
+> There is no absolute rules to categorize atoms, molecules and organisms, so, for me, I put complex assemblies or molecule repetition inside this category.
+
+> In fact, some components have more than three hierarchy assembly levels like you can have in Russian dolls analogy. So, I think it's relevant to communicate between team designer and developers to know what is the best category.
+
+![Organism: header](https://raw.githubusercontent.com/Gnuk/blog-usa/atomic-design-in-practice/images/2019/10/atomic-organisms.png)
+
+## Header by example
+
+The **header** organism represents the **logo** atom with the **main navigation** molecule and the **search form** molecule assemblies.
+
+> We can notice molecules are aligned to the bottom and the main navigation is extended. These properties should be described in our organism.
+
+### HTML (Pug)
+
+```pug
+.organism-header
+    .organism-header-logo
+        +atom-logo
+    .organism-header-navigation
+        +molecule-navigation
+    .organism-header-form-search
+        +molecule-form-search
+```
+
+### CSS (SCSS)
+
+```scss
+.organism-header {
+  display: flex;
+  > [class^="organism-header"] {
+    margin-right: 40px;
+    align-self: flex-end;
+    &:last-child {
+      margin-right: 0;
+    }
+  }
+  .organism-header-navigation {
+    flex-grow: 1;
+  }
+  .organism-header-form-search {
+    width: 250px;
+  }
+}
+```
+
+# Templates and pages
+
+![The template on the left side and the page, the instance, on the right side](https://raw.githubusercontent.com/Gnuk/blog-usa/atomic-design-in-practice/images/2019/10/atomic-templates-pages.png)
+
+Templates represents the page structure. This step is just a canvas to put our organisms. The template will be the model of our page, so the page is an instance of a template.
+
+![Template and page example](https://raw.githubusercontent.com/Gnuk/blog-usa/atomic-design-in-practice/images/2019/10/atomic-templates-pages-example.png)
```

Comment:
If you are the one making the templates, in the right section "What is?", I think it would be better to say "What is it?"

Reply:
I've lost the image, it will be difficult to me to replace it ^^"